pkg/flock/flock_unix.go

```diff
@@ -30,7 +35,28 @@ func Acquire(path string) (int, error) {
 	return lock, unix.Flock(lock, unix.LOCK_EX)
 }
 
+// AcquireShared creates a shared lock on a file for the duration of the process, or until Release(d).
+// This method is reentrant.
+func AcquireShared(path string) (int, error) {
+	lock, err := unix.Open(path, unix.O_CREAT|unix.O_RDWR, 0600)
+	if err != nil {
+		return -1, err
+	}
+	return lock, unix.Flock(lock, unix.LOCK_SH)
+}
+
 // Release removes an existing lock held by this process.
 func Release(lock int) error {
 	return unix.Flock(lock, unix.LOCK_UN)
 }
+
+// CheckLock checks whether any process is using the lock
+func CheckLock(path string) bool {
+	lockByte, _ := exec.Command("lsof", "-w", "-F", "ln", path).Output()
+	locks := string(lockByte)
+	if locks == "" {
+		return false
+	}
+	readWriteLock := strings.Split(locks, "\n")[2]
```

Comment:
Is there a chance this could come back without the expected indexes?

Reply:
Unlikely, I am explicitly forcing the exact number of output fields with the "-F", so if the lock isn't empty, it will contain those fields.
Command output looks like this
```
$ lsof -w -F ln /tmp/testlock.test 
p27002
f7
lW
n/tmp/testlock.test
```
Where "lW" indicates a write lock